quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/integration/web/QuarkusRequestFilter.java

```diff
@@ -90,6 +93,10 @@ private void configureEndHandler(RoutingContext context, Promise<Object> promise
                 close(session);
             } catch (Throwable cause) {
                 promise.fail(cause);
+                context.response().headers().clear();
+                context.response().putHeader(HttpHeaderNames.CONTENT_TYPE, HttpHeaderValues.TEXT_PLAIN);
+                context.response().putHeader(HttpHeaderNames.CONTENT_LENGTH, "0");
+                context.response().setStatusCode(HttpResponseStatus.INTERNAL_SERVER_ERROR.code());
```

Comment:
Would it not be better to throw this to let the ErrorHandler handle it since that has loads of login on how to handle different exceptions, and return correct response if it should be html or json?

Reply:
Even for the Wildfly distribution, the response does not go through the error handler. And the response you get in Wildfly is the Wildfly error page, including with the stacktrace.

In Quarkus, we are not leaking any details and just returning a `500` response.